js/data/bucket/circle_bucket.js

```diff
@@ -40,19 +40,9 @@ CircleBucket.prototype.programInterfaces = {
             components: 4,
             type: 'Uint8',
             isLayerConstant: false,
-            value: (
-                'this._premultiplyColor(' +
-                    'layer.getPaintValue("circle-color", globalProperties, featureProperties),' +
-                    'layer.getPaintValue("circle-opacity", globalProperties, featureProperties)' +
-                ')'
-            ),
+            value: 'this._premultiplyColor(layer.getPaintValue("circle-color", globalProperties, featureProperties), 1)',
```

Comment:
Yes, to multiply the alpha component of the color into the rgb components. After we split up color and opacity attributes for all layer types I think we'll be able to drop the second parameter from `premultiplyColor`.